gickup: init at 0.10.18

rhousand opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ x ] x86_64-linux
  - [ ] aarch64-linux
  - [ x ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gickup</li>
  </ul>
</details>

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

rhousand committed:
maintainers: add rhousand

rhousand committed:
gickup: init at 0.10.17

fabaff commented on pkgs/tools/backup/gickup/default.nix:
```diff
@@ -0,0 +1,25 @@
+    owner = "cooperspencer";
+    repo = "gickup";
+    rev = "v${version}";
+    sha256 = "sha256-tiQmb7bBWb99k23lS+d+YR14y4YeYPWqccl/2DLv7Dk=";
```
Nowadays, using the SRI hash is the default.

```suggestion
    hash = "sha256-tiQmb7bBWb99k23lS+d+YR14y4YeYPWqccl/2DLv7Dk=";
```

fabaff commented on pkgs/tools/backup/gickup/default.nix:
```diff
@@ -0,0 +1,25 @@
+    sha256 = "sha256-tiQmb7bBWb99k23lS+d+YR14y4YeYPWqccl/2DLv7Dk=";
+  };
+
+  vendorSha256 = "sha256-DWGrs/ZKMKgVfwU7W+dktLELbW9Co7cmDy9pWVP5p2w=";
```
```suggestion
  vendorHash = "sha256-DWGrs/ZKMKgVfwU7W+dktLELbW9Co7cmDy9pWVP5p2w=";
```

fabaff commented on pkgs/tools/backup/gickup/default.nix:
```diff
@@ -0,0 +1,25 @@
+  vendorSha256 = "sha256-DWGrs/ZKMKgVfwU7W+dktLELbW9Co7cmDy9pWVP5p2w=";
+
+  meta = with lib; {
+    description = "Backup all your repositories with Ease.";
```
```suggestion
    description = "Tool to backup repositories";
```

No period at the end (was mentioned in the past in CONTRIBUTING.md), the rest is opinionated.

fabaff commented on pkgs/tools/backup/gickup/default.nix:
```diff
@@ -0,0 +1,25 @@
+
+  meta = with lib; {
+    description = "Backup all your repositories with Ease.";
+    homepage = "https://github.com/cooperspencer/gickup";
```
Tools (e.g., `nix-update`) could pick up the changelog data and add it to the commit messages and the PR description for package updates.

```suggestion
    homepage = "https://github.com/cooperspencer/gickup";
    changelog = "https://github.com/cooperspencer/gickup/releases/tag/v${version}";
```

rhousand committed:
gickup: init at 0.10.17

Changed sha256 to hash and vendorSha256 to vendorHash.
Removed period from end of description.
Added changelog url

rhousand committed:
Merge branch 'NixOS:master' into pkg-gickup

rhousand committed:
Merge branch 'NixOS:master' into pkg-gickup

rhousand committed:
gickup: init at 0.10.18

adamcstephens reviewed (changes requested):
output binary has no version, please add `ldflags = ["-X main.version=${version}"];`

https://nixos.org/manual/nixpkgs/stable/#ssec-go-common-attributes

adamcstephens commented:
Please also rebase/rewrite your commit history to two commits. Adding yourself as maintainer, and init at 0.10.18

rhousand committed:
Merge branch 'NixOS:master' into pkg-gickup

rhousand committed:
Merge branch 'NixOS:master' into pkg-gickup

adamcstephens commented:
0.10.20 is out, and the feedback suggested should be addressed.
